### PR TITLE
Feature/scheduler validation

### PR DESCRIPTION
**Pull request by adibas03**

### CODE REVIEW

File: app/stores/index.js [hunk 1]
```diff
@@ -18,5 +18,6 @@ export const stores = {
   routing: routingStore,
   transactionStore,
   //scheduleStore
-  scheduleStore:scheduleStores
+  scheduleStore:scheduleStores,
+  services
```
Comment (Bagaric) [inline app/stores/index.js]: Why adding `services` to the `store` constant? There already is a constant named `services` in `app/services/index.js`, is it not?

File: app/components/ScheduleWizard/AbstractSetting.js [hunk 2]
```diff
@@ -10,20 +11,101 @@ class AbstractSetting extends Component {
...
+      _validationsErrors[property] = errors[errorState-1];
+    }
+    return _validations[property];
+  }
+
+	onChange = (name) => (event)=> {
+		const { target } = event;
+		const { scheduleStore } = this.props;
+		scheduleStore[name] = target.value;
+    this.validate(name)(event);
```
Comment (Bagaric) [inline app/components/ScheduleWizard/AbstractSetting.js]: Weird indentation.

File: app/components/Common/ScheduleWizard.js [hunk 3]
```diff
@@ -12,10 +12,84 @@ class ScheduleWizard extends Component {
...
+      timeBounty:'',
+      deposit:''
+    },
+    InfoSettings:{
+      toAddress:'',
+      gasAmount:'',
+      amountToSend:'',
+      gasPrice:'',
+      yourData:''
+    }
```
Comment (Bagaric) [inline app/components/Common/ScheduleWizard.js]: Maybe adding spaces after `:`? Looks a bit cleaner.

File: app/components/ScheduleWizard/InfoSettings.js [hunk 4]
```diff
@@ -3,77 +3,101 @@ import { inject, observer } from 'mobx-react';
...
+      this.toggleYourData = this.toggleYourData.bind(this);
     }
-    componentDidMount() {
+
+    validators = {
+      toAddress:this.ethereumAddressValidator(),
+      gasAmount:this.integerValidator(),
+      amountToSend:this.decimalValidator(),
+      gasPrice:this.integerValidator(),
+      yourData:{
```
Comment (Bagaric) [inline app/components/ScheduleWizard/InfoSettings.js]: Maybe adding spaces after `:`? Looks a bit cleaner.

File: app/stores/index.js (hunk 1 above)
Comment (adibas03) [reply app/stores/index.js]: it is not exposed, presently.
This exposes it, so we could have access to it.

Comment (Bagaric) [reply app/stores/index.js]: Are you sure it's not exposed? It has the `export` keyword, same as this `stores` const. https://github.com/chronologic/eth-alarm-clock-dapp/blob/feature/scheduler_validation/app/services/index.js#L7

Comment (adibas03) [reply app/stores/index.js]: I mean as a provider, since we3Services saves some state in web3

Comment (adibas03) [reply app/stores/index.js]: Oh, never mind

Comment (adibas03) [reply app/stores/index.js]: Thanks @Bagaric

**Comment by adibas03** [comment]
@Bagaric Pushed updates. Final review ??

**Comment by Bagaric** [comment]
@adibas03 Looking good :) great work!

**Pull request merged**